subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -183,44 +190,61 @@ public void setupProjectLayout() {
-            if (initDescriptor == null) {
-                ComponentType componentType = inputHandler.selectOption("Select type of project to generate", projectLayoutRegistry.getComponentTypes(), projectLayoutRegistry.getDefault().getComponentType());
-                List<Language> languages = projectLayoutRegistry.getLanguagesFor(componentType);
-                if (languages.size() == 1) {
-                    initDescriptor = projectLayoutRegistry.get(componentType, languages.get(0));
-                } else {
-                    if (!languages.contains(Language.JAVA)) {
-                        // Not yet implemented
-                        throw new UnsupportedOperationException();
-                    }
-                    Language language = inputHandler.selectOption("Select implementation language", languages, Language.JAVA);
-                    initDescriptor = projectLayoutRegistry.get(componentType, language);
-                }
-            }
-        } else {
-            initDescriptor = projectLayoutRegistry.get(type);
+        BuildInitializer initDescriptor = getBuildInitializer(inputHandler, projectLayoutRegistry);
+
+        ModularizationOption modularizationOption = getModularizationOption(inputHandler, initDescriptor);
+
+        BuildInitDsl dsl = getBuildInitDsl(inputHandler, initDescriptor);
+
+        boolean useIncubatingAPIs = shouldUseIncubatingAPIs(inputHandler);
+
+        BuildInitTestFramework testFramework = getBuildInitTestFramework(inputHandler, initDescriptor, modularizationOption);
+
+        String projectName = getProjectName(inputHandler, initDescriptor);
+
+        String packageName = getPackageName(inputHandler, initDescriptor, projectName);
+
+        java.util.Optional<JavaLanguageVersion> toolChainVersion = getJavaLanguageVersion(inputHandler, initDescriptor.getLanguage());
+
+        validatePackageName(packageName);
+
+        List<String> subprojectNames = initDescriptor.getComponentType().getDefaultProjectNames();
+        InitSettings settings = new InitSettings(
+            projectName,
+            useIncubatingAPIs,
+            subprojectNames,
+            modularizationOption,
+            dsl,
+            packageName,
+            testFramework,
+            insecureProtocol.get(),
+            projectDir,
+            toolChainVersion);
+        initDescriptor.generate(settings);
+
+        initDescriptor.getFurtherReading(settings).ifPresent(link -> getLogger().lifecycle("Get more help with your project: {}", link));
+    }
+
+    java.util.Optional<JavaLanguageVersion> getJavaLanguageVersion(UserInputHandler inputHandler, Language language) {
+        if(isJvmLanguage(language)) {
+            JavaLanguageVersion current = JavaLanguageVersion.of(Jvm.current().getJavaVersion().getMajorVersion());
+            return of(inputHandler.selectOption("Select a Java toolchain to build the project", DefaultJavaLanguageVersion.getKnownVersions().collect(toList()), current));
```

Comment:
❗ I think the formatting of the version selection message needs to be changed.  With a long list, there should be more of a distinction between the option to select and the JDK version number - for the later version especially this is confusing.  They should perhaps say "JDK 11, JDK 12, etc."

<img width="400" alt="image" src="https://user-images.githubusercontent.com/44197/210397243-bd8eaf17-84ac-4bd9-b125-82154a84cce8.png">